proxy/api/src/service.rs

```diff
@@ -39,27 +46,56 @@ impl Environment {
     ///
     /// If `test_mode` is `true` then `Environment::temp_dir` is set for temporary on-disk
     /// persistence.
-    fn new(test_mode: bool) -> Result<Self, Error> {
+    fn new(test_mode: bool, test_use_rad_home: bool) -> Result<Self, Error> {
         if test_mode {
-            let temp_dir = tempfile::tempdir()?;
-            let coco_profile = coco::profile::Profile::from_root(temp_dir.path(), None)?;
-            let keystore = Arc::new(coco::keystore::memory());
-            Ok(Self {
-                key: None,
-                temp_dir: Some(temp_dir),
-                coco_profile,
-                keystore,
-                test_mode,
-            })
+            if test_use_rad_home {
+                let root_path = match std::env::var_os("RAD_HOME") {
+                    None => {
+                        println!("RAD_HOME must be set when running with --test-use-rad-home");
+                        std::process::exit(1);
+                    },
+                    Some(root) => path::Path::new(&root).to_path_buf(),
+                };
+                let store_path = root_path.join("store");
+                let coco_profile =
+                    coco::profile::Profile::from_root(path::Path::new(&root_path), None)?;
+                let keystore = Arc::new(coco::keystore::file(coco_profile.paths().clone()));
```

Comment:
This is exactly the behavior when the `--test` flag is not provided. I think we don’t need the additional `--test-use-rad-home` flag and can just run the test node with `RAD_HOME` set.

Reply:
Makes sense. Will all the control API endpoints work though? We don't need them for this integration test, but maybe for some future ones?

Reply:
Removed it in: b58fbc64a31d03ad13495d01838d4480f4829915.